perception/object_recognition/tracking/multi_object_tracker/CMakeLists.txt

```diff
@@ -1,34 +1,29 @@
-cmake_minimum_required(VERSION 3.0.2)
+cmake_minimum_required(VERSION 3.5)
 project(multi_object_tracker)
 
-add_compile_options(-std=c++14)
+### Compile options
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wno-unused-parameter -Wall -Wextra -Wpedantic -Werror)
+endif()
 
-find_package(catkin REQUIRED COMPONENTS
-  autoware_perception_msgs
-  roscpp
-  tf2
-  tf2_ros
-  unique_id
-  uuid_msgs
-)
+### Find Packages
+find_package(ament_cmake_auto REQUIRED)
 
-find_package(Eigen3 REQUIRED)
+## Find Eigen Dependencies
+find_package(eigen3_cmake_module REQUIRED)
+find_package(Eigen3)
 
-catkin_package(
-  CATKIN_DEPENDS
-    autoware_perception_msgs
-    roscpp
-    tf2
-    tf2_ros
-    unique_id
-    uuid_msgs
-)
+ament_auto_find_build_dependencies()
 
-include_directories(include ${catkin_INCLUDE_DIRS} ${EIGEN3_INCLUDE_DIR})
+# includes
+include_directories(SYSTEM ${EIGEN3_INCLUDE_DIR})
```

Comment:
Without this line I was not able to find the eigen headers, the error output below is returned when that line is removed:

```
jilada@ade:~/AutowareArchitectureProposal (ros2-porting/add-ublox-deps $% u=)$ colcon build --packages-up-to multi_object_tracker
Starting >>> autoware_perception_msgs
Finished <<< autoware_perception_msgs [0.58s]
Starting >>> multi_object_tracker
--- stderr: multi_object_tracker
In file included from /home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/perception/object_recognition/tracking/multi_object_tracker/include/multi_object_tracker/multi_object_tracker_core.hpp:22,
                 from /home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/perception/object_recognition/tracking/multi_object_tracker/src/multi_object_tracker_core.cpp:20:
/home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/perception/object_recognition/tracking/multi_object_tracker/include/multi_object_tracker/data_association/data_association.hpp:26:10: fatal error: Eigen/Core: No such file or directory
   26 | #include <Eigen/Core>
      |          ^~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/multi_object_tracker_node.dir/build.make:63: CMakeFiles/multi_object_tracker_node.dir/src/multi_object_tracker_core.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/perception/object_recognition/tracking/multi_object_tracker/src/tracker/model/bicycle_tracker.cpp:28:10: fatal error: Eigen/Core: No such file or directory
   28 | #include <Eigen/Core>
      |          ^~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/multi_object_tracker_node.dir/build.make:115: CMakeFiles/multi_object_tracker_node.dir/src/tracker/model/bicycle_tracker.cpp.o] Error 1
In file included from /home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/perception/object_recognition/tracking/multi_object_tracker/src/data_association/data_association.cpp:20:
/home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/perception/object_recognition/tracking/multi_object_tracker/include/multi_object_tracker/data_association/data_association.hpp:26:10: fatal error: Eigen/Core: No such file or directory
   26 | #include <Eigen/Core>
      |          ^~~~~~~~~~~~
compilation terminated.
/home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/perception/object_recognition/tracking/multi_object_tracker/src/tracker/model/pedestrian_tracker.cpp:28:10: fatal error: Eigen/Core: No such file or directory
   28 | #include <Eigen/Core>
      |          ^~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/multi_object_tracker_node.dir/build.make:141: CMakeFiles/multi_object_tracker_node.dir/src/data_association/data_association.cpp.o] Error 1
make[2]: *** [CMakeFiles/multi_object_tracker_node.dir/build.make:128: CMakeFiles/multi_object_tracker_node.dir/src/tracker/model/pedestrian_tracker.cpp.o] Error 1
/home/jilada/AutowareArchitectureProposal/src/autoware/autoware.iv/perception/object_recognition/tracking/multi_object_tracker/src/tracker/model/vehicle_tracker.cpp:28:10: fatal error: Eigen/Core: No such file or directory
   28 | #include <Eigen/Core>
      |          ^~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/multi_object_tracker_node.dir/build.make:102: CMakeFiles/multi_object_tracker_node.dir/src/tracker/model/vehicle_tracker.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:107: CMakeFiles/multi_object_tracker_node.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make: *** [Makefile:141: all] Error 2
---
Failed   <<< multi_object_tracker [3.59s, exited with code 2]

```

Reply:
Does `target_link_libraries(my_target Eigen3::Eigen)` like described [here](https://index.ros.org/doc/ros2/Tutorials/Ament-CMake-Documentation/#adding-dependencies) instead of `
include_directories(SYSTEM ${EIGEN3_INCLUDE_DIR})` work?

Reply:
That works excellently, I just push my changes: 0b8b53d